app/controllers/orders_controller.rb

```diff
@@ -1,48 +1,75 @@
 class OrdersController < ApplicationController
-  before_action :find_order, only: [:cart, :checkout, :update, :receipt]
-  before_action :check_access, only: [:cart, :checkout, :receipt]
-
+  before_action :set_order, only: [:cart, :checkout, :add_to_cart, :update, :receipt]
+  before_action :set_seller_order, only: [:show]
+  before_action :set_product, only: [:add_to_cart]
+  before_action :set_seller, only: [:index, :show]
+  before_action :require_seller_login, only: [:index, :show]
 
   def cart; end
```

Comment:
Nice! This is the best way, according to our style guide, to use methods that have no body